File: fsharp/GameOfLife.FSharp.Engine/PopulationPatterns.fs
namespace GameOfLife.FSharp.Engine

module PopulationPatterns =

    /// <summary>
    /// Blinker Pattern - Period 2
    /// </summary>
    let blinker = PopulationPattern.create "Blinker" (array2D [|
        [| Dead; Alive; Dead |];
        [| Dead; Alive; Dead |];
        [| Dead; Alive; Dead |];
    |])

    /// <summary>
    /// Toad Pattern - Period 2
    /// </summary>v
    let toad = PopulationPattern.create "Toad" (array2D [|
        [| Dead; Dead; Dead; Dead |];
        [| Dead; Alive; Alive; Alive |];
        [| Alive; Alive; Alive; Dead |];
        [| Dead; Dead; Dead; Dead |];
    |])

    /// <summary>
    /// Beacon Pattern - Period 2
    /// </summary>
    let beacon = PopulationPattern.create "Beacon" (array2D [|
        [| Alive; Alive; Dead; Dead |];
        [| Alive; Alive; Dead; Dead |];
        [| Dead; Dead; Alive; Alive |];
        [| Dead; Dead; Alive; Alive |];
    |])

    /// <summary>
    /// Pulsar Pattern - Period 3
    /// </summary>
    let pulsar = PopulationPattern.create "Pulsar" (array2D [|
        [| Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead |];
        [| Dead; Dead; Dead; Alive; Alive; Alive; Dead; Dead; Dead; Alive; Alive; Alive; Dead; Dead; Dead |];
        [| Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead |];
        [| Dead; Alive; Dead; Dead; Dead; Dead; Alive; Dead; Alive; Dead; Dead; Dead; Dead; Alive; Dead |];
        [| Dead; Alive; Dead; Dead; Dead; Dead; Alive; Dead; Alive; Dead; Dead; Dead; Dead; Alive; Dead |];
        [| Dead; Alive; Dead; Dead; Dead; Dead; Alive; Dead; Alive; Dead; Dead; Dead; Dead; Alive; Dead |];
        [| Dead; Dead; Dead; Alive; Alive; Alive; Dead; Dead; Dead; Alive; Alive; Alive; Dead; Dead; Dead |];
        [| Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead |];
        [| Dead; Dead; Dead; Alive; Alive; Alive; Dead; Dead; Dead; Alive; Alive; Alive; Dead; Dead; Dead |];
        [| Dead; Alive; Dead; Dead; Dead; Dead; Alive; Dead; Alive; Dead; Dead; Dead; Dead; Alive; Dead |];
        [| Dead; Alive; Dead; Dead; Dead; Dead; Alive; Dead; Alive; Dead; Dead; Dead; Dead; Alive; Dead |];
        [| Dead; Alive; Dead; Dead; Dead; Dead; Alive; Dead; Alive; Dead; Dead; Dead; Dead; Alive; Dead |];
        [| Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead |];
        [| Dead; Dead; Dead; Alive; Alive; Alive; Dead; Dead; Dead; Alive; Alive; Alive; Dead; Dead; Dead |];
        [| Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead; Dead |];
    |])
    
    let all = [|
        blinker; toad; beacon; pulsar
    |]

File: csharp/GameOfLife.CSharp.Engine.Tests/PopulationPatterns.cs
using GameOfLife.CSharp.Engine;
using System.Collections.Generic;

namespace GameOfLife.Engine
{
    public static class PopulationPatterns
    {
        private const bool dead = false;
        private const bool live = true;

        /// <summary>
        /// A collection of predefined patterns: Blinker, Toad, Beacon, Pulsar
        /// </summary>
        public static ICollection<PopulationPattern> AllPatterns => new[]
        {
            Blinker,
            Toad,
            Beacon,
            Pulsar
        };

        /// <summary>
        /// Blinker Pattern - Period 2
        /// </summary>
        public static PopulationPattern Blinker => PopulationPattern.FromArray2D(1, "Blinker", new[,]
        {
            { dead, live, dead },
            { dead, live, dead },
            { dead, live, dead }
        });

        /// <summary>
        /// Toad Pattern - Period 2
        /// </summary>
        public static PopulationPattern Toad => PopulationPattern.FromArray2D(2, "Toad", new[,]
        {
            { dead, dead, dead, dead },
            { dead, live, live, live },
            { live, live, live, dead },
            { dead, dead, dead, dead }
        });

        /// <summary>
        /// Beacon Pattern - Period 2
        /// </summary>
        public static PopulationPattern Beacon => PopulationPattern.FromArray2D(3, "Beacon", new[,]
        {
            { live, live, dead, dead },
            { live, live, dead, dead },
            { dead, dead, live, live },
            { dead, dead, live, live }
        });

        /// <summary>
        /// Pulsar Pattern - Period 3
        /// </summary>
        public static PopulationPattern Pulsar => PopulationPattern.FromArray2D(4, "Pulsar", new[,]
        {
            { dead, dead, dead, dead, dead, dead, dead, dead, dead, dead, dead, dead, dead, dead, dead },
            { dead, dead, dead, live, live, live, dead, dead, dead, live, live, live, dead, dead, dead },
            { dead, dead, dead, dead, dead, dead, dead, dead, dead, dead, dead, dead, dead, dead, dead },
            { dead, live, dead, dead, dead, dead, live, dead, live, dead, dead, dead, dead, live, dead },
            { dead, live, dead, dead, dead, dead, live, dead, live, dead, dead, dead, dead, live, dead },
            { dead, live, dead, dead, dead, dead, live, dead, live, dead, dead, dead, dead, live, dead },
            { dead, dead, dead, live, live, live, dead, dead, dead, live, live, live, dead, dead, dead },
            { dead, dead, dead, dead, dead, dead, dead, dead, dead, dead, dead, dead, dead, dead, dead },
            { dead, dead, dead, live, live, live, dead, dead, dead, live, live, live, dead, dead, dead },
            { dead, live, dead, dead, dead, dead, live, dead, live, dead, dead, dead, dead, live, dead },
            { dead, live, dead, dead, dead, dead, live, dead, live, dead, dead, dead, dead, live, dead },
            { dead, live, dead, dead, dead, dead, live, dead, live, dead, dead, dead, dead, live, dead },
            { dead, dead, dead, dead, dead, dead, dead, dead, dead, dead, dead, dead, dead, dead, dead },
            { dead, dead, dead, live, live, live, dead, dead, dead, live, live, live, dead, dead, dead },
            { dead, dead, dead, dead, dead, dead, dead, dead, dead, dead, dead, dead, dead, dead, dead },
        });
    }
}
